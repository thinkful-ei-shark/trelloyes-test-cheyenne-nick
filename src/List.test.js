import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';

import List from './List';


describe('List />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        const card = {id: 'a', title: 'test', content:'test content'}
        ReactDOM.render(<List key={1} header="test" cards={[card]}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    })
})



describe('<List />', () => {
    it('renders this UI as expected', () => {
        const card = {id: 'a', title: 'test', content:'test content'}
        // Render the component, as JSON
        const tree = renderer.create(<List key={1} header="test" cards={[card]} />).toJSON();
        // Check whether it matches the previous snapshot
        // Stored in __snapshots__/App.test.js.snap
        expect(tree).toMatchSnapshot(); 
    });
});