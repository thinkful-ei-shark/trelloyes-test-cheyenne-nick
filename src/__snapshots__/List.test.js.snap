// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders this UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      test
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        test
      </h3>
      <p>
        test content
      </p>
    </div>
  </div>
  <button
    className="List-add-button"
    type="button"
  >
    + Add Random Card
  </button>
</section>
`;
